actions/provision_errors.py

```diff
@@ -0,0 +1,195 @@
+import socket
+import six
+import st2client
+import st2client.commands.action
+import st2client.models
+from st2client.client import Client
+from lib.base_action import BaseAction
+
+
+QUEUED_STATUSES = [
+    st2client.commands.action.LIVEACTION_STATUS_REQUESTED,
+    st2client.commands.action.LIVEACTION_STATUS_SCHEDULED,
+    st2client.commands.action.LIVEACTION_STATUS_DELAYED,
+]
+
+PROGRESS_STATUSES = [
+    st2client.commands.action.LIVEACTION_STATUS_PAUSING,
+    st2client.commands.action.LIVEACTION_STATUS_PAUSED,
+    st2client.commands.action.LIVEACTION_STATUS_RESUMING,
+    st2client.commands.action.LIVEACTION_STATUS_RUNNING
+]
+
+COMPLETED_STATUSES = [
+    st2client.commands.action.LIVEACTION_STATUS_SUCCEEDED
+]
+
+ERRORED_STATUSES = [
+    st2client.commands.action.LIVEACTION_STATUS_FAILED,
+    st2client.commands.action.LIVEACTION_STATUS_TIMED_OUT,
+    st2client.commands.action.LIVEACTION_STATUS_ABANDONED,
+    st2client.commands.action.LIVEACTION_STATUS_CANCELED
+]
+
+SERVICENOW_STATUSES = {
+    'queued': QUEUED_STATUSES,
+    'running': PROGRESS_STATUSES,
+    'succeeded': COMPLETED_STATUSES,
+    'failed': ERRORED_STATUSES
+}
+
+PROVISION_SKIP_LIST = [
+    'vm_delete',
+    'provision_cleanup',
+    'provision_cleanup_exec'
+]
+
+COMPLETED_STATUSES = [
+    st2client.commands.action.LIVEACTION_STATUS_SUCCEEDED
+]
+
+
+class ErrorsProvisionErrors(BaseAction):
+
+    def __init__(self, config):
+        """Creates a new BaseAction given a StackStorm config object (kwargs works too)
+        :param config: StackStorm configuration object for the pack
+        :returns: a new BaseAction
+        """
+        super(ErrorsProvisionErrors, self).__init__(config)
+        self.st2_exe_id = ""
+
+    def check_status(self, st2_execution):
+
+        st2_status = 'unknown'
+        for k, v in six.iteritems(SERVICENOW_STATUSES):
+            if st2_execution and st2_execution.status in v:
+                st2_status = k
+
+        # st2_execution_comments comments needs to be defaulted to an empty string
+        # else there is a trigger error saying that the value does not exists.
+        execution_status = {
+            'st2_execution_id': self.st2_exe_id,
+            'st2_execution_status': st2_status,
+            'st2_execution_comments': ""
+        }
+
+        if st2_status == 'failed':
+            execution_status['st2_execution_comments'] = self.get_st2_error(st2_execution)
+        elif st2_status == 'unknown':
+            execution_status['st2_execution_comments'] = ("Could not find execution_id "
+                                                        "in database")
+
+        return execution_status
+
+    def get_st2_error(self, st2_execution):
```

Comment:
Instead of doing things this way like it was in the Encore pack. I want to use the new methods (its better) that were created and used here: https://github.com/EncoreTechnologies/stackstorm-errors/blob/master/actions/get_formatted_error.py. The stuff here may need to be moved to the common base file so that they can be used in both places.. I would envision almost having a common `find_error` method that both of them call.

Reply:
This goes for lines 85 through 184